CDCgov/prime-reportstream
# 1585 Enhance serializers to provide proper errors/warnings with incorrect payloads

This PR adds proper error and warning messages when incorrect data is ingested by the serializers.

Test:
1. Run the unit tests and verify that the tests pass.
2. Send an HL7 message to a CSV sender and verify that the reportItemCount is 0 and that a warning or error is reported.
3. Send a CSV message to an HL7 sender and verify that the reportItemCount is 0 and that a warning or error is reported.

## Changes
- Errors in HL7 ingest will return a null report - same as the CSV serializer
- Refactored HL7 parsing to provide better error messages
- Added warning for empty CSV files

## Checklist

### Testing

### Process

## Fixes
- #1585

## To Be Done
